plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -137,7 +142,9 @@ public static Object jsonValueToTrinoValue(Type type, @Nullable Object jsonValue
         if (type == DateType.DATE) {
             return LocalDate.parse((String) jsonValue).toEpochDay();
         }
-        if (type == TIMESTAMP_MILLIS) {
+        if (type instanceof TimestampType timestampType) {
+            int precision = timestampType.getPrecision();
+            checkArgument(precision == 3 || precision == 6, "Unsupported precision: %s", precision);
             return Instant.parse((String) jsonValue).toEpochMilli() * MICROSECONDS_PER_MILLISECOND;
```

Comment:
`Instant.parse((String) jsonValue).toEpochMilli()` seems wrong for precision=6

Reply:
Spark always writes timestamp stats with millisecond precision. Left a code comment and asking in https://delta-users.slack.com/archives/CJ70UCSHM/p1716970335747839

Reply:
for stats millis precision is fine (provided that we never take them as actual min/max)

however, the code here is truncating to millis, which is not correct for max value. it should round up